### PR TITLE
JobIntentService replacement with WorkManager

### PR DESCRIPTION
**Pull request by zaghdaneh**
## Issue

This fixes the following issue(s):
- Resolves: [#1485](https://github.com/TUM-Dev/Campus-Android/issues/1485)

## Approach
- Changed the three classes that extended JobIntentService (Location, geofencing and Silence services).
Tried to keep the original code as untouched as possible while using the necessary new functions.

## Testing
- Tested both the silence and Location services (used by the Calendar) by trying random actions.
Was unable to find a way to test the geofencing Service.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt:83]:
```suggestion
```

File: app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt:96]:
```suggestion
}

```

File: app/src/main/java/de/tum/in/tumcampusapp/service/QueryLocationWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/QueryLocationWorker.kt:75]:
```suggestion
```

File: app/src/main/java/de/tum/in/tumcampusapp/service/QueryLocationWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/QueryLocationWorker.kt:72]:
```suggestion
```

File: app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt:137]: As above: is there a reason for keeping `@JvmStatic`?
```suggestion
        fun enqueueWork(context: Context) {
```

File: app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt:178]:
```suggestion
}

```

File: app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt:144]: as above: is `@JvmStatic` needed?
```suggestion
        fun dequeueWork(context: Context){
```

File: app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt:46]: I think the values which are later used coud be better defaults.
What do you think?

```suggestion
        val latitude = inputData.getDouble(Const.ADD_GEOFENCE_LAT, 48.137430)
        val longitude = inputData.getDouble(Const.ADD_GEOFENCE_LON, 11.575490)
        val range = inputData.getFloat(Const.ADD_GEOFENCE_RANGE, 50_000f)
```

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Sorry for the late review. It has been a rocky few weeks for me.

I have left some mild refactoring/clarification requests, but otherwise this looks very good.
I think this can be merged after the discussions I have left are cleared.

**Comment by CommanderStorm** [comment]
Also, please fix the linting issues in https://github.com/TUM-Dev/Campus-Android/actions/runs/5259629203/jobs/9534184369#step:3:1008 ^^

File: app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt
Comment (zaghdaneh) [reply app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt:74]: Not really, Since I was basically transforming a class into another, I just tried to make only the necessary change and avoid removing stuff I'm not sure what the consequences of are.

File: app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt
Comment (zaghdaneh) [reply app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt:46]: Good idea since it's for now the only used numbers by the geofencer. 0 as default value doesn't really make sense in this  situation when I think about it.

File: app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt
Comment (zaghdaneh) [reply app/src/main/java/de/tum/in/tumcampusapp/service/SilenceWorker.kt:38]: If it's removed, shouldn't it be part of a different pull request that removes all "old build support"? 
I would just keep it personally but will follow your call.

File: app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt
Comment (CommanderStorm) [reply app/src/main/java/de/tum/in/tumcampusapp/service/GeoFencingRegistrationWorker.kt]: `@JvmStatic` is a migration class when migrating java -> kotlin (simplifed: allows java code to clall kotlin code).
Given that `startGeofencing` is only called from kotlin this can be removed

**Comment by CommanderStorm** [comment]
Thanks for this PR.

For the next time: 
if you accept a comment with one of these buttons, this will automatically close said comment (thus not making me check if this was implemented) 😉 
image

**Review by CommanderStorm — APPROVED** [review]

**Pull request merged**
